services/explorer-indexer/src/modules/historical-sync/historical.syncer.service.ts

```diff
@@ -219,21 +219,28 @@ export default class HistoricalSyncService implements OnModuleInit {
       this.syncState.number < this.chainState.number;
       this.syncState.number++
     ) {
-      chuncks.push({
+      if (chunks.length === 1000) {
```

Comment:
Are you sure the problem comes from here and not this:
```
const foundBlockNumbers = await this.pbftService.getSavedPbftPeriods();
```

?

Reply:
Yes, I was able to reproduce the problem via my local setup by doing:

- Exported the dump from the QA database.
- Imported locally
- Set the 2Gig limit  in a local docker container

So then, the error we had was there and this one fixed it and I was running the producer constantly and rerunning it repeatedly for the latter part of the day.

However, that can also be optimized, I'll add a QB query for that too. Thanks for pointing it out 👍🏼

Reply:
Okay so I optimized this one too to return only the `pbft.number` property and tested on 13047 PBFT entries:

- [ ] node --max-old-space-size=2048 dist/src/main
- [ ] node --max-old-space-size=1024 dist/src/main
- [ ] node --max-old-space-size=512 dist/src/main
- [ ] node --max-old-space-size=256 dist/src/main
- [ ] node --max-old-space-size=128 dist/src/main
- [ ] node --max-old-space-size=64 dist/src/main
 
At 64 MB it fails the chunk creation because of the heap memory, but we should be good to go when it comes to database reads as that passed even at 64MB

Reply:
I think we can improve this further. 13k PBFT blocks is a small number.
The mainnet has 758k blocks after running for 1 month.

See comments on the PR.